src/tests/ftest/util/command_utils.py

```diff
@@ -167,6 +168,30 @@ def with_exports(self):
         """
         return command_as_user(self.with_bind, self.run_user, self.env)
 
+    @contextlib.contextmanager
+    def temp_exit_status_exception(self, tmp_value):
+        """Set exit_status_exception temporarily.
+
+        Args:
+            tmp_value (bool): the temporary value to set
+        """
+        original_value = self.exit_status_exception
+        self.exit_status_exception = tmp_value
+        yield
+        self.exit_status_exception = original_value
```

Comment:
I've also wondered if this should be more generic, like:
```
    @contextlib.contextmanager
    def temp_attr(self, attr_name, tmp_value):
        """Set a class attribute temporarily.

        Args:
            attr_name (str): class attribute to set
            tmp_value (bool): the temporary value to set

        Raises:
            AttributeError: if the attribute does not exist
        """
        original_value = get_attr(self, attr_name)
        setattr(self, attr_name, tmp_value)
        yield
        setattr(self, attr_name, original_value)
```

Which could replace `temp_exit_status_exception`, `temp_run_user`, and any future attributes